niworkflows/interfaces/mni.py

```diff
@@ -231,18 +354,115 @@ def _validate_results(self):
+    image space and have the same dimensions.
+    """
     import nibabel as nb
     import os
-
+    # Load the input image
     in_nii = nb.load(in_file)
+    # Load the mask image
     mask_nii = nb.load(mask_file)
+    # Set all non-mask voxels in the input file to zero.
     data = in_nii.get_data()
     data[mask_nii.get_data() == 0] = 0
+    # Save the new masked image.
     new_nii = nb.Nifti1Image(data, in_nii.affine, in_nii.header)
     new_nii.to_filename(new_name)
     return os.path.abspath(new_name)
 
 
+def create_cfm(in_file, lesion_mask=None, global_mask=True, out_path=None):
+    """
+    Create a mask to constrain registration.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to an existing image (usually a mask).
+        If global_mask = True, this is used as a size/dimension reference.
+    out_path : str
+        Path/filename for the new cost function mask.
+    lesion_mask : str, optional
+        Path to an existing binary lesion mask.
+    global_mask : bool
+        Create a whole-image mask (True) or limit to reference mask (False)
+
+    Returns
+    -------
+    str
+        Absolute path of the new cost function mask.
+
+    Notes
+    -----
+    in_file and lesion_mask must be in the same
+    image space and have the same dimensions
+    """
+    import os
+    import numpy as np
+    import nibabel as nb
+    from nipype.utils.filemanip import fname_presuffix
+
+    if out_path is None:
+        out_path = fname_presuffix(in_file, suffix='_cfm', newpath=os.getcwd())
+    else:
+        out_path = os.path.abspath(out_path)
+
+    if not global_mask and not lesion_mask:
+        NIWORKFLOWS_LOG.warning(
+            'No lesion mask was provided and global_mask not requested, '
+            'therefore the original mask will not be modified.')
+
+    # Load the input image
+    in_img = nb.load(in_file)
+
+    # If we want a global mask, create one based on the input image.
+    if global_mask:
+        # Create a mask of ones with the shape of the input image.
+        data = np.ones(in_img.shape, dtype=np.uint8)
+    else:
+        data = in_img.get_data()
+        if set(np.unique(data)) - {0, 1}:
+            raise ValueError("`global_mask` must be true if `in_file` is not a binary mask")
+
+    # If a lesion mask was provided, combine it with the secondary mask.
+    if lesion_mask is not None:
+        # Reorient the lesion mask and get the data.
+        lm_img = nb.as_closest_canonical(nb.load(lesion_mask))
+
+        # Subtract lesion mask from secondary mask, set negatives to 0
+        data = np.fmax(data - lm_img.get_data(), 0)
+        # Cost function mask will be created from subtraction
+    # Otherwise, CFM will be created from global mask
+
+    cfm_img = nb.Nifti1Image(data, in_img.affine, in_img.header)
+
+    # Save the cost function mask.
+    cfm_img.set_data_dtype(np.uint8)
+    cfm_img.to_filename(out_path)
+
+    return out_path
```

Comment:
At this point, if `global_mask=True` and `lesion_mask = None` then this mask will be all-ones. It'd be easier to debug if we returned `None` and didn't set a mask into ANTs. WDYT?

Reply:
So just return [here](https://github.com/poldracklab/niworkflows/pull/233/files#diff-34ffe363a301e0c280a14f9fb630ac2fR433)? Can do that.

Reply:
Masks have to come in pairs. When this is called in that way, it's because we need a mask with null-effect to partner with the intended mask.